corehq/apps/geospatial/tests/test_utils.py

```diff
@@ -116,3 +117,48 @@ def test_set_user_gps_property(self):
         set_user_gps_property(self.DOMAIN, submit_data)
         user = CommCareUser.get_by_user_id(self.user.user_id, self.DOMAIN)
         self.assertEqual(user.get_user_data(self.DOMAIN)[GPS_POINT_CASE_PROPERTY], '1.23 4.56 0.0 0.0')
+
+
+class TestUpdateCasesOwner(TestCase):
+    domain = 'test-domain'
+
+    def setUp(self):
+        super().setUp()
+        self.user_a = CommCareUser.create(self.domain, 'User_A', '1234', None, None)
+        self.case_1 = create_case(self.domain, case_id=uuid4().hex, save=True, owner_id=self.user_a.user_id)
+        self.case_2 = create_case(self.domain, case_id=uuid4().hex, save=True, owner_id=self.user_a.user_id)
+
+        self.user_b = CommCareUser.create(self.domain, 'User_B', '1234', None, None)
+        self.case_3 = create_case(self.domain, case_id=uuid4().hex, save=True, owner_id=self.user_b.user_id)
+        self.case_4 = create_case(self.domain, case_id=uuid4().hex, save=True, owner_id=self.user_b.user_id)
+
+    def tearDown(self):
+        self.user_a.delete(self.domain, None)
+        self.user_b.delete(self.domain, None)
+        CommCareCase.objects.hard_delete_cases(
+            self.domain,
+            [self.case_1.case_id, self.case_2.case_id, self.case_3.case_id, self.case_4.case_id]
+        )
```

Comment:
>  I think SQL objects should be automatically removed by tearDown as the transaction would be rolled back after every test run.

Good point.  This is the expected behaviour.
Although I was not able to get it working for `CommcareCase` . I tested for a different model `GeoPolygon` and it is being deleted after the test run. Maybe because CommcareCase is partitioned . 
Keeping the delete code for now.

Another pattern I have observed on HQ that SQL objects are deleted explicitly in `teardown`.  Wondering if there is reason for this ?

Reply:
I am not sure if it has to do anything with partitioning 🤔  and I don't think we partition the test data (maybe I am wrong here)

> Keeping the delete code for now.

👍 

> Another pattern I have observed on HQ that SQL objects are deleted explicitly in teardown. Wondering if there is reason for this ?

I don't think it should be a practice and there is any reasoning behind it. SQL objects should be automatically cleared in the tests. Maybe there are usecases where you might have to clear them manually but it should work out of the box.

Reply:
>  I don't think it should be a practice and there is any reasoning behind it. SQL objects should be automatically cleared in the tests. Maybe there are usecases where you might have to clear them manually but it should work out of the box.

Agreed.  I think it should be implicitly understood by the very nature of how Django's `TestCase` works.  Good point to keep in mind for me for future work